urls.go

```diff
@@ -201,6 +201,19 @@ func normalizeEscape(s string) (string, error) {
 	return escape(u), nil
 }
 
+// parserest parase the content after the first symbol ':'.
+// If there is a port after ':',  ("", url) is returned. e.g. "www.abc.com:80/"
+// (url[:i], url[i+1:]) is returned in other stations.
+func parseRest(url string, i int) (scheme, path string) {
+	portReg := regexp.MustCompile(`^:(\d+)(/|$)`)
+	matched := portReg.MatchString(url[i:])
+	if matched {
```

Comment:
After we do everything above, I think it makes more sense to just fold this logic into getScheme itself. There's not much to this function at all, other than just being an if-statement.